File: leetcode/Climbing_Stairs.java
import java.util.*;

public class Solution {
    public int climbStairs(int n) {
        if (n <= 1) return 1;
        int[] c = new int[3];
        c[0] = c[1] = 1;
        for (int i = 2; i <= n; ++i) {
            c[2] = c[0] + c[1];
            c[0] = c[1];
            c[1] = c[2];
        }
        return c[1];
    }
}

File: leetcode/Climbing_Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> d(n + 1, 0);
        d[0] = 1;
        for (int i = 0; i < n; ++i) {
            d[i + 1] += d[i];
            if (i <= n - 2)
                d[i + 2] += d[i];
        }
        return d[n];
    }
};
